server/regoracle/server.go

```diff
@@ -0,0 +1,171 @@
+package regoracle
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/edgelesssys/ego/enclave"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/medibloc/panacea-doracle/server/service"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/medibloc/panacea-doracle/types"
+	log "github.com/sirupsen/logrus"
+	tos "github.com/tendermint/tendermint/libs/os"
+	"os"
+	"path/filepath"
+)
+
+var nodePrivKeyPath string
+
+type Server struct {
+	*service.Service
+}
+
+func NewServer(svc *service.Service) *Server {
+	return &Server{
+		Service: svc,
+	}
+}
+
+func (srv *Server) Run() error {
+	// if node key exists, continue the process of registration with the existing key
+	if tos.FileExists(nodePrivKeyPath) {
+		return srv.continueRegistration()
+	}
+
+	// if there is no node private key, create a new node key and request OracleRegistration
+	return srv.createRegistration()
+}
+
+func (srv *Server) Close() error {
+	return nil
+}
+
+// continueRegistration continues the oracle registration process using the existing node key
+func (srv *Server) continueRegistration() error {
+	// get existing node key
+	nodePrivKeyBz, err := sgx.UnsealFromFile(nodePrivKeyPath)
+	if err != nil {
+		return err
+	}
+	_, nodePubKey := crypto.PrivKeyFromBytes(nodePrivKeyBz)
+
+	// get unique ID
+	selfEnclaveInfo, err := sgx.GetSelfEnclaveInfo()
+	if err != nil {
+		return err
+	}
+
+	uniqueID := base64.StdEncoding.EncodeToString(selfEnclaveInfo.UniqueID)
+
+	// get oracle registration from Panacea.
+	// TODO: Should be replaced with light client query for data verification
```

Comment:
Thanks for marking TODO. 
I think we initialize query client with `NewQueryClient` at `query_client.go` , before send query to panacea.
At this time, trusted block info, context, rpcAddr and chainId should be needed. So, I'm wondering where is the best place to call `NewQueryClient`.  Also, we should consider shut down of the query client  if oracle registration fails.

What about your opinion?

Reply:
The trusted block info would be given via flag of this command(`FlagTrustedBlockHeight`, `FlagTrustedBlockHash`).
And rpcAddr and chainID are added in `config.toml` as configuration params.

With our current structure, I would create a `NewQueryClient` inside `service.New` and the service would have a query client just like a gRPC client.